packages/core/pipes.spec.ts

```diff
@@ -19,6 +20,12 @@ describe("core pipes", () => {
     expect(result).toEqual([1, 2, 3]);
   });
 
+  it("JSONP pipe should throw BadRequestException when given json is invalid", () => {
+    expect(() => {
+      JSONP.transform('*{"key":"value"}', undefined);
+    }).toThrow(new HttpException("Unexpected token * in JSON at position 0", 400));
```

Comment:
@htuna07 Is it required to have the exact same string in the HttpException or does `toThrow` checks only the type?

Reply:
The given value to `toThrow()` is optional. If value exists, it compares the expected value and actual value strictly. Without any given value, it only checks thrown something or not.

But the `toThrowError()` can check the type of error without comparing expected and actual value.

Reply:
Based on your answer, i understand that it would be enough to use

`toThrow(new HttpException("some string", 400));`

Is that correct?

Reply:
Yes, just like I did. The other option is ;
`toThrowError(HttpException,"some string")`. But there is no chance to compare **statusCode** in this option.